src/osemgrep/reporting/Summary_report.ml

```diff
@@ -15,8 +15,29 @@ module Out = Semgrep_output_v1_t
 (* Entry point *)
 (*****************************************************************************)
 
+exception FoundGitDir of Fpath.t
+
+let find_git_dir lst =
+  try
+    let _ =
+      List.fold_left
+        (fun _ x ->
+          let dir =
+            if Common2.dir_exists (Fpath.to_string x) then x else Fpath.parent x
+          in
+          if Git_wrapper.is_git_repo dir then raise (FoundGitDir dir))
+        () lst
+    in
+    None
+  with
+  | FoundGitDir x -> Some x
```

Comment:
Initially I did just use `List.iter` but I had a shower thought concern that `is_git_repo` could take quite a bit of time if we have a large list of targets. 

I'm not very familiar with the expected (or extreme cases of) lengths of inputs, and I thought exiting early was desirable. Also, I thought about a parallel map reduce but that's out of my current ocaml knowledge scope 😅

Reply:
`List.find_opt` and `List.exists` will both only go up to the first occurrence! Depending on implementation, they may still traverse the rest of the list, but they won't run `is_git_repo` on the rest.

Similarly, with the fold I wrote, since it checks the accumulator first, it won't run `is_git_repo` more than it needs to.

Reply:
I think with your approach `List.iter` would be equivalent to `List.fold`! Since you exit early through the exception either way